osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs

```diff
@@ -95,19 +96,36 @@ protected override ResultsScreen CreateResults(ScoreInfo score)
             return new TimeshiftResultsScreen(score, roomId.Value.Value, playlistItem, true);
         }
 
-        protected override ScoreInfo CreateScore()
+        protected override Score CreateScore()
         {
             var score = base.CreateScore();
-            score.TotalScore = (int)Math.Round(ScoreProcessor.GetStandardisedScore());
+            score.ScoreInfo.TotalScore = (int)Math.Round(ScoreProcessor.GetStandardisedScore());
+            return score;
+        }
+
+        protected override async Task SubmitScore(Score score)
+        {
+            await base.SubmitScore(score);
 
             Debug.Assert(token != null);
 
-            var request = new SubmitRoomScoreRequest(token.Value, roomId.Value ?? 0, playlistItem.ID, score);
-            request.Success += s => score.OnlineScoreID = s.ID;
-            request.Failure += e => Logger.Error(e, "Failed to submit score");
-            api.Queue(request);
+            var tcs = new TaskCompletionSource<bool>();
+            var request = new SubmitRoomScoreRequest(token.Value, roomId.Value ?? 0, playlistItem.ID, score.ScoreInfo);
 
-            return score;
+            request.Success += s =>
```

Comment:
we could really do with a better way of asyncing this in the future, if this becomes a more common flow (`api.Perform` should return this awaitable task).